sl_model_mod.f90

```diff
@@ -161,6 +166,8 @@ subroutine sl_call_readnl
                         initial_topo, iceVolume, coupling, patch_ice, &
                         L_sim, dt1, dt2, dt3, dt4, Ldt1, Ldt2, Ldt3, Ldt4, whichplanet)
 
+      call sl_get_model_res(norder, nglv)
```

Comment:
From what I can see, this is the only place `sl_get_model_res` is called.  If that's true, then it seems like it would be much simpler to just have these as additional arguments to `sl_readnl`.  Why do these need a separate subroutine?

Reply:
You are right in that that's the only place the subroutine is called, and lumping the call within `sl_readnl` is possible/cleaner. The choice was made due to the calling of the subroutine made in `li_bedtopo` module on the MALI side for coupled simulations (https://github.com/MALI-Dev/E3SM/blob/77688ad717e5f159dea24d4d86a624245937418a/components/mpas-albany-landice/src/mode_forward/mpas_li_bedtopo.F#L478). MALI needs to know what `nglv` value is being used in the SLM to allocate dimension to the SLM-related variables, but it doesn't need to know any other fields defined in `namelist.sealevel`.